packages/wallet/ui/src/util/WalletBackendAdapter.js

```diff
@@ -1,26 +1,29 @@
 import { E } from '@endo/eventual-send';
+import { makeFollower, mapAsyncIterable, iterateLatest } from '@agoric/casting';
 import { Far } from '@endo/marshal';
-import {
-  makeAsyncIterableFromNotifier,
-  makeNotifierKit,
-} from '@agoric/notifier';
+import { makeNotifierKit } from '@agoric/notifier';
 
 const newId = kind => `${kind}${Math.random()}`;
 
-/**
- * @template T
- * @param {ERef<Notifier<T>>} notifier
- */
-const iterateNotifier = notifier =>
-  makeAsyncIterableFromNotifier(notifier)[Symbol.asyncIterator]();
-
-export const makeBackendFromWalletBridge = walletBridge => {
+export const makeBackendFromWalletBridge = (walletBridge, makeLeader) => {
   const { notifier: servicesNotifier } = makeNotifierKit(
     harden({
       board: E(walletBridge).getBoard(),
     }),
   );
 
+  /**
+   * @template T
+   * @param {ERef<Notifier<T>>} notifier
+   */
+  const iterateNotifier = notifier => {
```

Comment:
Is this something that we'd want to reuse across dapps?

Reply:
Not really.  It's a hack to deal with notifiers.   @gibson042's new `makePublisherKit` will be the way forward (instead of the wallet's current `makeNotifierKit`), and that will have helpers in `@agoric/notifier` to consume them.   At that time, we will also update the `@agoric/casting` follower API to conform with them.